### DISCUSSION TITLE
Aircrack-ng bottle not available

### DISCUSSION THREAD

**Discussion by patriotarcher** [Homebrew/discussions#878] [Everyday usage]
I'm trying to install aircrack-ng on my Mac and on the homebrew website it says it supports macOS but when I try to 'brew install aircrack-ng' it says the bottle is unavailable. Any help? Is this something I did or something on homebrew's end? 
Thanks

**Comment by carlocab** [comment]
What does `brew config` and `brew doctor` say?

**Reply by patriotarcher** [reply]
brew doctor say ready to brew and brew config seems normal. Not sure what exactly I'm looking for in brew config

**Reply by carlocab** [reply]
Please paste the output of `brew config` here. I can't help you if I don't know what it says.

**Reply by SujanSapkota07** [reply]
brew install --build-from-source aircrack-ng
this line of command worked for my mac, try once,
to check whether it worked or not, i followed the following commands given in the link below, and i worked
https://martinsjean256.wordpress.com/2018/02/12/hacking-aircrack-ng-on-mac-cracking-wi-fi-without-kali-in-parallels/

**Comment by patriotarcher** [comment]
HOMEBREW_VERSION: 3.0.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: c951be8d3c7b339c7e759b7c40aec859e09a70a5
Last commit: 11 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 74ed7719621987660220cc6ba115ea2b8f24665f
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.30.1 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.1-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4
Rosetta 2: false

**Comment by carlocab** [comment]
`aircrack-ng` has no bottle for ARM yet. You can try building it from source with `brew install --build-from-source aircrack-ng`.

If that works, and you're able to use it, let me know and we can get it bottled.

**Reply by patriotarcher** [reply]
Then what's this mean?
<img width="672" alt="Screen Shot 2021-02-21 at 3 58 05 PM" src="https://user-images.githubusercontent.com/79422190/108640146-a5f4da80-745d-11eb-840c-3a86459665b5.png">

**Reply by gromgit** [reply]
Exactly what @carlocab said:
> `aircrack-ng` has no bottle for ARM yet

The only bottles currently available are for the four most recent releases of _Intel_ macOS.

**Reply by carlocab** [reply]
Try comparing it to a formula that has an ARM bottle. For example: https://formulae.brew.sh/formula/llvm#default

**Reply by jtreg** [reply]
no solution yet?

I get this:

(base) James-MacBook-Pro:aircrack-ng jtreg1$ brew install --build-from-source aircrack-ng
==> Downloading https://download.aircrack-ng.org/aircrack-ng-1.6.tar.gz
Already downloaded: /Users/jtreg1/Library/Caches/Homebrew/downloads/7fd41b718c0967eb5bbfcbeca9f654fbe327ace6ededffa25097547768f103e6--aircrack-ng-1.6.tar.gz
==> Patching
==> ./autogen.sh --disable-silent-rules --disable-dependency-tracking --prefix=/opt/homebrew/Cellar/aircrack-ng/1.6
==> make install
Last 15 lines from /Users/jtreg1/Library/Logs/Homebrew/aircrack-ng/02.make:
/bin/sh ./libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I.  -I./include/aircrack-ng/osdep -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6 -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap   -Wno-unused-command-line-argument  -I/include -I/usr/local/include -I/opt/local/include -I../.. -D_FORTIFY_SOURCE=2  -Wall -Wno-unused-command-line-argument -O3 -std=gnu99 -fno-strict-aliasing -Wpointer-arith -Wstrict-overflow=2 -Wstrict-prototypes -fvisibility=hidden -fstack-protector-strong -Wno-array-bounds  -I/include  -c -o lib/osdep/libaircrack_osdep_la-common.lo `test -f 'lib/osdep/common.c' || echo './'`lib/osdep/common.c
/bin/sh ./libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I.  -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6 -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap   -Wno-unused-command-line-argument  -I/include -I/usr/local/include -I/opt/local/include -I../.. -D_FORTIFY_SOURCE=2  -Wall -Wno-unused-command-line-argument -O3 -std=gnu99 -fno-strict-aliasing -Wpointer-arith -Wstrict-overflow=2 -Wstrict-prototypes -fvisibility=hidden -fstack-protector-strong -Wno-array-bounds  -I/include  -c -o lib/radiotap/radiotap.lo lib/radiotap/radiotap.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I./include/aircrack-ng/osdep -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6 -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -Wno-unused-command-line-argument -I/include -I/usr/local/include -I/opt/local/include -I../.. -D_FORTIFY_SOURCE=2 -Wall -Wno-unused-command-line-argument -O3 -std=gnu99 -fno-strict-aliasing -Wpointer-arith -Wstrict-overflow=2 -Wstrict-prototypes -fvisibility=hidden -fstack-protector-strong -Wno-array-bounds -I/include -c lib/osdep/darwin_tap.c  -fno-common -DPIC -o lib/osdep/.libs/libaircrack_osdep_la-darwin_tap.o
/bin/sh ./libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I.  -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6 -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap   -Wno-unused-command-line-argument  -I/include -I/usr/local/include -I/opt/local/include -I../.. -D_FORTIFY_SOURCE=2 -pthread   -I/opt/homebrew/Cellar/pcre/8.45/include "-DLIBAIRCRACK_CE_WPA_PATH=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/\"" "-DABS_TOP_SRCDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\"" "-DABS_TOP_BUILDDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\"" "-DLIBDIR=\"/opt/homebrew/Cellar/aircrack-ng/1.6/lib\""   -Wall -Wno-unused-command-line-argument -O3 -std=gnu99 -fno-strict-aliasing -Wpointer-arith -Wstrict-overflow=2 -Wstrict-prototypes -fvisibility=hidden -fstack-protector-strong -Wno-array-bounds  -I/include  -c -o lib/libac/adt/libaircrack_la-avl_tree.lo `test -f 'lib/libac/adt/avl_tree.c' || echo './'`lib/libac/adt/avl_tree.c
/bin/sh ./libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I.  -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6 -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap   -Wno-unused-command-line-argument  -I/include -I/usr/local/include -I/opt/local/include -I../.. -D_FORTIFY_SOURCE=2 -pthread   -I/opt/homebrew/Cellar/pcre/8.45/include "-DLIBAIRCRACK_CE_WPA_PATH=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/\"" "-DABS_TOP_SRCDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\"" "-DABS_TOP_BUILDDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\"" "-DLIBDIR=\"/opt/homebrew/Cellar/aircrack-ng/1.6/lib\""   -Wall -Wno-unused-command-line-argument -O3 -std=gnu99 -fno-strict-aliasing -Wpointer-arith -Wstrict-overflow=2 -Wstrict-prototypes -fvisibility=hidden -fstack-protector-strong -Wno-array-bounds  -I/include  -c -o lib/libac/adt/libaircrack_la-circular_buffer.lo `test -f 'lib/libac/adt/circular_buffer.c' || echo './'`lib/libac/adt/circular_buffer.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I./include/aircrack-ng/osdep -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6 -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -Wno-unused-command-line-argument -I/include -I/usr/local/include -I/opt/local/include -I../.. -D_FORTIFY_SOURCE=2 -Wall -Wno-unused-command-line-argument -O3 -std=gnu99 -fno-strict-aliasing -Wpointer-arith -Wstrict-overflow=2 -Wstrict-prototypes -fvisibility=hidden -fstack-protector-strong -Wno-array-bounds -I/include -c lib/osdep/common.c  -fno-common -DPIC -o lib/osdep/.libs/libaircrack_osdep_la-common.o
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6 -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -Wno-unused-command-line-argument -I/include -I/usr/local/include -I/opt/local/include -I../.. -D_FORTIFY_SOURCE=2 -Wall -Wno-unused-command-line-argument -O3 -std=gnu99 -fno-strict-aliasing -Wpointer-arith -Wstrict-overflow=2 -Wstrict-prototypes -fvisibility=hidden -fstack-protector-strong -Wno-array-bounds -I/include -c lib/radiotap/radiotap.c  -fno-common -DPIC -o lib/radiotap/.libs/radiotap.o
/bin/sh ./libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I.  -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6 -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap   -Wno-unused-command-line-argument  -I/include -I/usr/local/include -I/opt/local/include -I../.. -D_FORTIFY_SOURCE=2 -pthread   -I/opt/homebrew/Cellar/pcre/8.45/include "-DLIBAIRCRACK_CE_WPA_PATH=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/\"" "-DABS_TOP_SRCDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\"" "-DABS_TOP_BUILDDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\"" "-DLIBDIR=\"/opt/homebrew/Cellar/aircrack-ng/1.6/lib\""   -Wall -Wno-unused-command-line-argument -O3 -std=gnu99 -fno-strict-aliasing -Wpointer-arith -Wstrict-overflow=2 -Wstrict-prototypes -fvisibility=hidden -fstack-protector-strong -Wno-array-bounds  -I/include  -c -o lib/libac/adt/libaircrack_la-circular_queue.lo `test -f 'lib/libac/adt/circular_queue.c' || echo './'`lib/libac/adt/circular_queue.c
20 errors generated.
make[1]: *** [src/aircrack-ng/aircrack_ng-linecount.o] Error 1
make[1]: *** Waiting for unfinished jobs....
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6 -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -Wno-unused-command-line-argument -I/include -I/usr/local/include -I/opt/local/include -I../.. -D_FORTIFY_SOURCE=2 -pthread -I/opt/homebrew/Cellar/pcre/8.45/include -DLIBAIRCRACK_CE_WPA_PATH=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/\" -DABS_TOP_SRCDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\" -DABS_TOP_BUILDDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\" -DLIBDIR=\"/opt/homebrew/Cellar/aircrack-ng/1.6/lib\" -Wall -Wno-unused-command-line-argument -O3 -std=gnu99 -fno-strict-aliasing -Wpointer-arith -Wstrict-overflow=2 -Wstrict-prototypes -fvisibility=hidden -fstack-protector-strong -Wno-array-bounds -I/include -c lib/libac/adt/avl_tree.c  -fno-common -DPIC -o lib/libac/adt/.libs/libaircrack_la-avl_tree.o
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6 -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -Wno-unused-command-line-argument -I/include -I/usr/local/include -I/opt/local/include -I../.. -D_FORTIFY_SOURCE=2 -pthread -I/opt/homebrew/Cellar/pcre/8.45/include -DLIBAIRCRACK_CE_WPA_PATH=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/\" -DABS_TOP_SRCDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\" -DABS_TOP_BUILDDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\" -DLIBDIR=\"/opt/homebrew/Cellar/aircrack-ng/1.6/lib\" -Wall -Wno-unused-command-line-argument -O3 -std=gnu99 -fno-strict-aliasing -Wpointer-arith -Wstrict-overflow=2 -Wstrict-prototypes -fvisibility=hidden -fstack-protector-strong -Wno-array-bounds -I/include -c lib/libac/adt/circular_buffer.c  -fno-common -DPIC -o lib/libac/adt/.libs/libaircrack_la-circular_buffer.o
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6 -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/include -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -I/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/lib/radiotap -Wno-unused-command-line-argument -I/include -I/usr/local/include -I/opt/local/include -I../.. -D_FORTIFY_SOURCE=2 -pthread -I/opt/homebrew/Cellar/pcre/8.45/include -DLIBAIRCRACK_CE_WPA_PATH=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6/\" -DABS_TOP_SRCDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\" -DABS_TOP_BUILDDIR=\"/private/tmp/aircrack-ng-20211116-61545-akgrr9/aircrack-ng-1.6\" -DLIBDIR=\"/opt/homebrew/Cellar/aircrack-ng/1.6/lib\" -Wall -Wno-unused-command-line-argument -O3 -std=gnu99 -fno-strict-aliasing -Wpointer-arith -Wstrict-overflow=2 -Wstrict-prototypes -fvisibility=hidden -fstack-protector-strong -Wno-array-bounds -I/include -c lib/libac/adt/circular_queue.c  -fno-common -DPIC -o lib/libac/adt/.libs/libaircrack_la-circular_queue.o
make: *** [install-recursive] Error 1

READ THIS: https://docs.brew.sh/Troubleshooting

(base) James-MacBook-Pro:aircrack-ng jtreg1$

**Reply by carlocab** [reply]
What you posted about what you get answers your question, no? If you'd like to try fixing this, you can try reporting the build failure to the aircrack-ng developers, in case they aren't aware of it already.
